### PR TITLE
add tests for Odata metadata document

### PR DESCRIPTION
**Pull request by nickpell**
First step in adding tests for the Odata api.

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]

File: corehq/apps/api/odata/tests/test_metadata.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        response = self._execute_metadata_query(wrong_credentials)
+        self.assertEqual(response.status_code, 401)
+
+    def test_missing_feature_flag(self):
+        correct_credentials = self._get_correct_credentials()
+        response = self._execute_metadata_query(correct_credentials)
+        self.assertEqual(response.status_code, 404)
+
+    def test_no_case_types(self):
+        self._enable_odata_toggle_for_domain()
```
Comment (czue) [inline corehq/apps/api/odata/tests/test_metadata.py]: nit: I think could use `@flag_enabled` decorator for this

File: corehq/apps/api/odata/tests/data/empty_metadata_document.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+    <edmx:DataServices>
+        <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CommCare">
+            <EntityContainer Name="Container" >
+            </EntityContainer>
```
Comment (czue) [inline corehq/apps/api/odata/tests/data/empty_metadata_document.xml]: realizing this is just exposing existing behavior (which is great! hooray tests!) but it seems a little odd to me that we wouldn't include the built in meta properties even if you have no custom properties defined. is that by design?

**Review by czue — APPROVED** [review]
no blockers

File: corehq/apps/api/odata/tests/data/empty_metadata_document.xml (hunk 2 above)
Comment (nickpell) [reply corehq/apps/api/odata/tests/data/empty_metadata_document.xml]: The point of this test is for when there are no case types, so we wouldn't include the meta properties, I think.

File: corehq/apps/api/odata/tests/test_metadata.py (hunk 1 above)
Comment (nickpell) [reply corehq/apps/api/odata/tests/test_metadata.py]: Thanks for the tip.  Updated: 53b98d7f317

File: corehq/apps/api/odata/tests/data/empty_metadata_document.xml (hunk 2 above)
Comment (czue) [reply corehq/apps/api/odata/tests/data/empty_metadata_document.xml]: oh whoops sorry i completely misread the test.

**Review by czue — APPROVED** [review]

**Pull request merged**